File: backend/Backend.Api/Controllers/AuthController.cs
using Backend.Aplication.InputModels;
using Backend.Aplication.Services.Interfaces;
using Backend.Aplication.ViewModels;
using Backend.Domain.Services;
using Backend.Domain.Validators;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace Backend.Api.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    [Authorize]
    public class AuthController : ControllerBase
    {
        private readonly IAuthService _service;
        public AuthController(IAuthService service) => _service = service;

        [HttpPost("login")]
        [AllowAnonymous]
        public async Task<ResultValidator<LoginViewModel>> Login([FromBody] LoginInputModel input) =>
            await _service.Login(input);

        [HttpGet("refresh-token/{input}")]
        public async Task<ResultValidator<LoginViewModel>> RefreshToken(string input) =>
            await _service.RefreshToken(input);

    }
}

File: frontend/src/controllers/authController.ts
import { ISigninInputModels, ISignupInputModels } from "../types/inputModels";
import { ILoginViewModel } from "../types/viewModels";
import { api } from "../service/api";
import { resolveResponse } from "../utils";

const route = 'Auth'

const Signin = (input: ISigninInputModels) => 
  new Promise<ILoginViewModel>((resolve,reject)=>
    api.post(`${route}/login`, input).then(({data})=> resolveResponse<ILoginViewModel>(resolve,reject,data)).catch(reject))

const RefreshToken = (input: string) => 
  new Promise<ILoginViewModel>((resolve,reject)=>
    api.get(`${route}/refresh-token/${input}`).then(({data})=> resolveResponse<ILoginViewModel>(resolve,reject,data)).catch(reject))

export const authController = {
  Signin,
  RefreshToken
}
